src/utils/upload/processUploadV2.js

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable no-param-reassign */
+import _ from 'lodash';
+
+import axios from 'axios';
+
+import { createSample, createSampleFileV2, updateSampleFileUploadV2 } from 'redux/actions/samples';
+
+import UploadStatus from 'utils/upload/UploadStatus';
+import loadAndCompressIfNecessary from 'utils/upload/loadAndCompressIfNecessary';
+import { inspectFile, Verdict } from 'utils/upload/fileInspector';
+
+// const putInS3 = async (loadedFileData, signedUrl, sampleUuid, fileType, onUploadProgress) => (
+const putInS3 = async (loadedFileData, signedUrl, onUploadProgress) => (
+  await axios.request({
+    method: 'put',
+    url: signedUrl,
+    data: loadedFileData,
+    headers: {
+      'Content-Type': 'application/octet-stream',
+    },
+    onUploadProgress,
+  }));
+
+const prepareAndUploadFileToS3 = async (
+  projectId, sampleId, fileType, file, signedUrl, dispatch,
+) => {
+  let loadedFile = null;
+
+  try {
+    loadedFile = await loadAndCompressIfNecessary(file, () => {
+      dispatch(updateSampleFileUploadV2(projectId, sampleId, fileType, UploadStatus.COMPRESSING));
+    });
+  } catch (e) {
+    const fileErrorStatus = e.message === 'aborted' ? UploadStatus.FILE_READ_ABORTED : UploadStatus.FILE_READ_ERROR;
+
+    dispatch(updateSampleFileUploadV2(projectId, sampleId, fileType, fileErrorStatus));
+    return;
+  }
+
+  try {
+    await putInS3(loadedFile, signedUrl, (progress) => {
+      const percentProgress = Math.round((progress.loaded / progress.total) * 100);
+
+      dispatch(
+        updateSampleFileUploadV2(
+          projectId, sampleId, fileType, UploadStatus.UPLOADING, percentProgress ?? 0,
+        ),
+      );
+    });
+  } catch (e) {
+    dispatch(updateSampleFileUploadV2(projectId, sampleId, fileType, UploadStatus.UPLOAD_ERROR));
+    return;
+  }
+
+  dispatch(updateSampleFileUploadV2(projectId, sampleId, fileType, UploadStatus.UPLOADED));
+};
+
+const fileTypes = {
+  matrix: 'matrix10x',
+  barcodes: 'barcodes10x',
+  features: 'features10x',
+  genes: 'features10x',
+};
+
+const getFileType = (fileName) => {
+  let fileType;
+
+  _.forEach(Object.entries(fileTypes), ([name, type]) => {
+    if (fileName.includes(name)) {
+      fileType = type;
+      return false;
+    }
+  });
```

Comment:
Not required to fix for this PR: is there a way to write this to have the 10x files format less hardcoded into the data upload function? It is something we will have to do in the near future

Reply:
As part of the UI refactoring to remove api v1 code I would like to make the fileTypes be identified and written correctly as soon as the files are dropped, so this piece of code won't be necessary anymore at that point (we'll store only fileType instead of fileName).

Reply:
ok, good!